src/document/__tests__/DocumentAnnotator-test.ts

```diff
@@ -273,4 +273,22 @@ describe('DocumentAnnotator', () => {
             expect(scrollToLocation).toHaveBeenCalledWith(parentEl, referenceEl, { offsets: { x: 15, y: 10 } });
         });
     });
+
+    describe('handleChangeMode()', () => {
+        beforeEach(() => {
+            annotator.annotatedEl = container.querySelector('.bp-doc') as HTMLElement;
+        });
+
+        test('should add and remove is highlighting class if mode changes', () => {
+            expect(annotator.annotatedEl?.classList.contains('ba-is-highlighting')).toBe(false);
+
+            annotator.emit(Event.ANNOTATIONS_MODE_CHANGE, { mode: Mode.HIGHLIGHT });
```

Comment:
We also want to test if `toggleAnnotationModeAction` emits the event, right? So instead of emitting the event directly, should we call `annotator.toggleAnnotationMode`?

Reply:
Good idea I'll make that change